src/autopas/molecularDynamics/LJFunctorAVX.h

```diff
@@ -467,13 +468,35 @@ class LJFunctorAVX : public Functor<Particle, ParticleCell, typename Particle::S
     const __m256d cutoffMask = _mm256_cmp_pd(dr2, _cutoffsquare, _CMP_LE_OS);
 
     // This requires that dummy is zero (otherwise when loading using a mask the owned state will not be zero)
-    const __m256i ownedStateJ = remainderIsMasked
-                                    ? _mm256_castpd_si256(_mm256_maskload_pd(
-                                          reinterpret_cast<double const *>(&ownedStatePtr2[j]), _masks[rest - 1]))
-                                    : _mm256_load_si256(reinterpret_cast<const __m256i *>(&ownedStatePtr2[j]));
-    // This requires that dummy is the first entry in OwnershipState!
-    const __m256d dummyMask = _mm256_cmp_pd(_mm256_castsi256_pd(ownedStateJ), _zero, _CMP_NEQ_UQ);
+    std::array<unsigned char, 4> ownedStateJ{};
+    if constexpr (remainderIsMasked) {
+      switch (rest) {
+          // std::memcpy(ownedStateJ.data(), &ownedStatePtr2[j], rest);
+          // without the switch a call to std::memcpy is generated, which we do not want!
+          // using a switch the compiler knows what rest can be 1, 2 or 3, and can optimize things.
+        case 1:
+          std::memcpy(ownedStateJ.data(), &ownedStatePtr2[j], rest);
+          break;
+        case 2:
+          std::memcpy(ownedStateJ.data(), &ownedStatePtr2[j], rest);
+          break;
+        case 3:
+          std::memcpy(ownedStateJ.data(), &ownedStatePtr2[j], rest);
+          break;
+      }
+      //_mm256_maskload_epi64(reinterpret_cast<long long const *>(&ownedStatePtr2[j]), _masks[rest - 1]);
+    } else {
+      std::memcpy(ownedStateJ.data(), &ownedStatePtr2[j], 4);
+      //_mm256_load_si256(reinterpret_cast<const __m256i *>(&ownedStatePtr2[j]));
+    }
+    // This requires that OwnershipState::dummy is zero!
+    static_assert(static_cast<unsigned char>(OwnershipState::dummy) == 0,
+                  "OwnershipState::dummy has to have the value 0.");
+    // const int32_t dummyMask = reinterpret_cast<int &>(*ownedStateJ.data());
+    const __m256i ownershipStateJ = _mm256_set_epi64x(ownedStateJ[3], ownedStateJ[2], ownedStateJ[1], ownedStateJ[0]);
+    const __m256d dummyMask = _mm256_castsi256_pd(_mm256_cmp_pd(ownershipStateJ, _zero, _CMP_NEQ_OQ));
```

Comment:
1. If you do this multiple times make a function (that can be inlined!) out of it.
2. ownershipStateJ is a _m256i but then used as a _m256d argument and the cast is then too late

Reply:
done in https://github.com/AutoPas/AutoPas/pull/490/commits/93d3e63017ad9b9bc9f2f83b935ca4796779a3ab